### PR TITLE
properly handle external kube proxy case

### PR DESCRIPTION
**Pull request by deads2k**
@derekwaynecarr Can you see if this works properly for you?

@liggitt I'm still trying to see if there's a reasonable way for me write an integration test for the external kube case.

### CODE REVIEW

**Comment by smarterclayton** [comment]
Start two masters, have one talk to the other?

**Comment by smarterclayton** [comment]
Run a master in a container and have it talk to another master in a container in test-integration-docker?

**Comment by deads2k** [comment]
> Start two masters, have one talk to the other?

Tried this.  We share an etcd in integration tests, so the controllers went ape-shit.




> Run a master in a container and have it talk to another master in a container in test-integration-docker?

I'll look into this.

**Comment by derekwaynecarr** [comment]
After chatting with David, this LGTM, but will wait for @liggitt  to review in more detail.

File: pkg/cmd/server/api/validation/master.go [hunk 1]
```diff
@@ -69,10 +69,16 @@ func ValidateMasterConfig(config *api.MasterConfig) fielderrors.ValidationErrorL
 	if config.KubernetesMasterConfig != nil {
 		allErrs = append(allErrs, ValidateKubernetesMasterConfig(config.KubernetesMasterConfig).Prefix("kubernetesMasterConfig")...)
 	}
+	if (config.KubernetesMasterConfig == nil) && (len(config.MasterClients.ExternalKubernetesKubeConfig) == 0) {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("kubernetesMasterConfig", config.KubernetesMasterConfig, "either kubernetesMasterConfig or masterClients.externalKubernetesKubeConfig must have a value"))
+	}
```
Comment (liggitt) [inline pkg/cmd/server/api/validation/master.go]: is it invalid to give us externalKubernetesKubeConfig if you also gave us a kubernetesMasterConfig?

**Comment by liggitt** [comment]
question about validate, LGTM otherwise

File: pkg/cmd/server/api/validation/master.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/server/api/validation/master.go]: fixed.

**Comment by deads2k** [comment]
Added another bit of validation.  Not having a new test doesn't increase the tech debt (this fixes a bug, nothing new).  Derek was hoping to have an image built tonight, so I'll [merge] and come back tomorrow for the test.

**Pull request merged**